front/poke/temporal/activities.ts

```diff
@@ -40,3 +85,424 @@ export async function scrubDataSourceActivity({
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
+  const runs = await Run.findAll({
+    where: {
+      workspaceId: workspace.id,
+    },
+    include: [
+      {
+        model: App,
+        as: "app",
+        required: true,
+      },
+    ],
+  });
+
+  const chunkSize = 8;
+  const chunks: Run[][] = [];
+  for (let i = 0; i < runs.length; i += chunkSize) {
+    chunks.push(runs.slice(i, i + chunkSize));
+  }
+
+  await front_sequelize.transaction(async (t) => {
+    for (let i = 0; i < chunks.length; i++) {
+      const chunk = chunks[i];
+      if (!chunk) {
+        continue;
+      }
+      await Promise.all(
+        chunk.map((run) => {
+          return (async () => {
+            const res = await coreAPI.deleteRun({
+              projectId: run.app.dustAPIProjectId,
+              runId: run.dustRunId,
+            });
+            if (res.isErr()) {
+              throw new Error(
+                `Error deleting Run from Core: ${res.error.message}`
+              );
+            }
+            await run.destroy({ transaction: t });
+          })();
+        })
+      );
+    }
+  });
+}
+
+export async function deleteMembersActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
+  await front_sequelize.transaction(async (t) => {
+    await MembershipInvitation.destroy({
+      where: {
+        workspaceId: workspace.id,
+      },
+      transaction: t,
+    });
+
+    const memberships = await Membership.findAll({
+      where: {
+        workspaceId: workspace.id,
+      },
+    });
+
+    if (memberships.length === 1) {
+      // We also delete the user if it has no other workspace.
+      const membership = memberships[0];
+      const membershipsOfUser = await Membership.findAll({
+        where: {
+          userId: membership.userId,
+        },
+      });
+      if (membershipsOfUser.length === 1) {
+        const user = await User.findOne({
+          where: {
+            id: membership.userId,
+          },
+        });
+        if (user) {
+          await UserMetadata.destroy({
+            where: {
+              userId: user.id,
+            },
+            transaction: t,
+          });
+          await membership.destroy({ transaction: t });
+          await user.destroy({ transaction: t });
+        }
+      }
+    }
+
+    for (const membership of memberships) {
+      await membership.destroy({ transaction: t });
+    }
+  });
+}
+
+export async function deleteWorkspaceActivity({
+  workspaceId,
+}: {
+  workspaceId: string;
+}) {
+  const auth = await Authenticator.internalBuilderForWorkspace(workspaceId);
+  const workspace = auth.workspace();
+
+  if (!workspace) {
+    throw new Error("Could not find the workspace.");
+  }
+
```

Comment:
Aren't we missing the workspace's deletion in Core?

Reply:
What do we create on Core when we create a workspace??

Reply:
My bad I was referring to apps's projects which are deleted :+1: